python/python/lance/dataset.py

```diff
@@ -4774,6 +4776,19 @@ def write_dataset(
         and can be retrieved using read_transaction().
         If both `commit_message` and `properties` are provided, `commit_message` will
         override any "lance.commit.message" key in `properties`.
+    data_path_uris: list of str, optional
+        *Experimental*. Additional bucket URIs for registering in the manifest
+        during dataset creation. When provided in CREATE mode, these bucket URIs
+        will be registered in the manifest as available buckets for future use.
+        Only used in CREATE mode for manifest registration.
+        Example: ["s3://bucket2/data", "s3://bucket3/data"]
+    target_path_uri: str, optional
```

Comment:
I think this should also be a list instead of just a single value

Reply:
If its a list we need to add a target selector (hash etc) and i was thinking that could be an extension PR?

Reply:
or maybe its easy to add

Reply:
You mean do a hash distribution to write to each base? I think we can say we only support one item in the list initially, but the API should be set to list so we can extend in the future